runtime/methoddata/src/main/java/org/qbicc/runtime/methoddata/MethodData.java

```diff
@@ -0,0 +1,134 @@
+package org.qbicc.runtime.methoddata;
+
+public class MethodData {
+    public static final class MethodInfo {
+        private int fileNameIndex;
+        private int classNameIndex;
+        private int methodNameIndex;
+        private int methodDescIndex;
+
+        public MethodInfo(int fileIndex, int classIndex, int methodIndex, int methodDescIndex) {
+            this.fileNameIndex = fileIndex;
+            this.classNameIndex = classIndex;
+            this.methodNameIndex = methodIndex;
+            this.methodDescIndex = methodDescIndex;
+        }
+
+        public boolean equals(Object other) {
+            if (this == other) return true;
+            if (other == null || getClass() != other.getClass()) return false;
+            MethodInfo that = (MethodInfo) other;
+            return fileNameIndex == that.fileNameIndex
+                && classNameIndex == that.classNameIndex
+                && methodNameIndex == that.methodNameIndex
+                && methodDescIndex == that.methodDescIndex;
+        }
+
+        public int getFileNameIndex() {
+            return fileNameIndex;
+        }
+
+        public int getClassNameIndex() {
+            return classNameIndex;
+        }
+
+        public int getMethodNameIndex() {
+            return methodNameIndex;
+        }
+
+        public int getMethodDescIndex() {
+            return methodDescIndex;
+        }
+
+        @Override
+        public int hashCode() {
+            int result = fileNameIndex;
+            result = 31 * result + classNameIndex;
+            result = 31 * result + methodNameIndex;
+            result = 31 * result + methodDescIndex;
+            return result;
+        }
+    }
+
+    public static final class CodePoint {
+        private int minfoIndex;
+        private int lineNumber;
+        private int bcIndex;
+        private int inlinedAtIndex;
```

Comment:
Does this recursively reference another `CodePoint`?

Reply:
Yes. `inlinedAtIndex` is the index of the `CodePoint` (now `SourceCodeInfo`) of the method that has inlined the method represented by this `CodePoint`.